test/UnitTests/MSTest.Analyzers.UnitTests/PreferDisposeOverTestCleanupAnalyzerTests.cs

```diff
@@ -45,53 +45,142 @@ public ValueTask DisposeAsync()
             }
             """;
 
-        await VerifyCS.VerifyAnalyzerAsync(code);
+        await VerifyCS.VerifyCodeFixAsync(code, code);
     }
 
     public async Task WhenTestClassHasTestCleanup_Diagnostic()
     {
         string code = """
             using Microsoft.VisualStudio.TestTools.UnitTesting;
-
+            
             [TestClass]
             public class MyTestClass
             {
                 [TestCleanup]
                 public void [|MyTestCleanup|]()
+                {
+                    int x = 1;
+                }
+            }
+            """;
+        string fixedCode = """
+            using System;
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            
+            [TestClass]
+            public class MyTestClass
+            : IDisposable
+            {
+                public void Dispose()
+                {
+                    int x = 1;
+                }
+            }
+            """;
+        await VerifyCS.VerifyCodeFixAsync(code, fixedCode);
+    }
+
+    public async Task WhenTestClassHasTestCleanup_WithAnotherBaseClass_Diagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System;
+            
+            public class LocalBase{}
+
+            [TestClass]
+            public class MyTestClass : LocalBase
+            {
+                [TestCleanup]
+                public void [|MyTestCleanup|]()
+                {
+                }
+            }
+            """;
+        string fixedCode = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+            using System;
+
+            public class LocalBase{}
+            
+            [TestClass]
+            public class MyTestClass : LocalBase, IDisposable
+            {
+                public void Dispose()
                 {
                 }
             }
             """;
+        await VerifyCS.VerifyCodeFixAsync(code, fixedCode);
+    }
 
-        await VerifyCS.VerifyAnalyzerAsync(code);
+    public async Task WhenTestClassHasTestCleanup_AndHasDispose_Diagnostic()
```

Comment:
Are we sure about this one? what if the two bodies conflicts?

Reply:
user have to fix it, in old fix we did it like the with Amaury